### PR TITLE
Can now admin override failed auto-validation of beta files (bug 1172690)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1172690](https://bugzilla.mozilla.org/show_bug.cgi?id=1172690)

### CODE REVIEW

File: apps/devhub/tests/test_views.py [hunk 1]
```diff
@@ -2849,45 +2890,25 @@ def test_listed_beta_not_pass_validation_with_flag(self, mock_sign_file):
...
-        # Make sure that the additional files are beta
-        fle = File.objects.all().order_by('-id')[0]
-        eq_(fle.status, amo.STATUS_BETA)
-
-    def test_force_not_beta(self):
-        self.post(beta=False)
-        f = File.objects.all().order_by('-created')[0]
-        assert f.status == amo.STATUS_PUBLIC
+    @mock.patch('devhub.views.sign_file')
+    def test_listed_beta_not_pass_validation_with_flag_override(
```
Comment (Natim) [inline apps/devhub/tests/test_views.py]: test_listed_beta_do_not_pass_validation_with_flag_override ?

**Comment by magopian** [comment]
@kmaglione could you please give this a quick r?

File: apps/devhub/views.py [hunk 2]
```diff
@@ -1206,9 +1206,10 @@ def auto_sign_file(file_, is_beta=False):
                              'comments': 'automatic validation'})
             helper.handler.process_preliminary(auto_validation=True)
     elif is_beta:
-        # Beta won't be reviewed. Either they pass validation and are
-        # automatically reviewed and signed, either they aren't accepted.
-        if validation.passed_auto_validation:
+        # Beta won't be reviewed. Either they pass validation (or have an admin
+        # override) and are automatically reviewed and signed, either they
```
Comment (kmaglione) [inline apps/devhub/views.py]: s/either/or/

File: apps/devhub/tests/test_views.py [hunk 3]
```diff
@@ -2889,6 +2849,68 @@ def test_force_not_beta(self):
...
+        self.upload.update(
+            validation='{"notices": 2, "errors": 1, "messages": [],'
+                       ' "metadata": {}, "warnings": 1,'
+                       ' "signing_summary": {"trivial": 1, "low": 1,'
+                       '                     "medium": 0, "high": 0},'
+                       ' "passed_auto_validation": 0}')
+        assert self.addon.status == amo.STATUS_PUBLIC
+        self.post(beta=True, expected_status=403)
+        file_ = File.objects.all().order_by('-created')[0]
+        # File status didn't change and the file is signed.
```
Comment (kmaglione) [inline apps/devhub/tests/test_views.py]: s/File/Addon/

File: apps/devhub/tests/test_views.py [hunk 4]
```diff
@@ -2889,6 +2849,68 @@ def test_force_not_beta(self):
...
+        assert self.addon.reload().status == amo.STATUS_PUBLIC
+        assert file_.status == amo.STATUS_BETA
```
Comment (kmaglione) [inline apps/devhub/tests/test_views.py]: Shouldn't this end up disabled?

**Pull request merged**